tests/check_libtelemetry.c

```diff
@@ -157,6 +157,45 @@ START_TEST(record_create_severity_overflow)
 }
 END_TEST
 
+START_TEST(record_set_event_id)
+{
+        int ret;
+        char *event_id = "aaaaaa00000033333344444466666622";
+        char *result;
+        ret = tm_create_record(&ref, 1, "t/t/t", 2000);
+        ck_assert_msg(ret != -ECONNREFUSED,
+                      "Opt-out enabled. Opt in to run this test");
+        if (asprintf(&result, "%s: %s\n", TM_EVENT_ID_STR, event_id) < 0) {
+                return;
+        }
+        ck_assert_str_ne(ref->record->headers[TM_EVENT_ID], result);
+        ret = tm_set_event_id(ref, event_id);
+        ck_assert_int_eq(ret, 0);
+        ck_assert_str_eq(ref->record->headers[TM_EVENT_ID], result);
+        free(result);
+        create_teardown();
+}
+END_TEST
+
+START_TEST(record_set_event_id_invalid)
+{
+        int ret;
+        char *event_id = "aaaaaa000000333333444444666666ZZ";
+        char *result;
+        ret = tm_create_record(&ref, 1, "t/t/t", 2000);
+        ck_assert_msg(ret != -ECONNREFUSED,
+                      "Opt-out enabled. Opt in to run this test");
+        if (asprintf(&result, "%s: %s\n", TM_EVENT_ID_STR, event_id) < 0) {
+                return;
+        }
+        ret = tm_set_event_id(ref, event_id);
+        ck_assert_int_eq(ret, 1);
+        ck_assert_str_ne(ref->record->headers[TM_EVENT_ID], result);
+        free(result);
+        create_teardown();
+}
+END_TEST
+
```

Comment:
Should also create tests for short event_id, long event_id, empty event_id, and NULL *event_id.